ait/core/evr.py

```diff
@@ -127,43 +128,74 @@ def format_message(self, evr_hist_data):
                 specified format strings. This is usually a result of the
                 expected data length and the byte array length not matching.
         '''
-        formatter_info = {
-            's': (-1, str),
-            'c': (1, str),
-            'i': (4, lambda h: int(binascii.hexlify(h), 16)),
-            'd': (4, lambda h: int(binascii.hexlify(h), 16)),
-            'u': (4, lambda h: int(binascii.hexlify(h), 16)),
-            'f': (4, lambda h: float(binascii.hexlify(h), 16)),
-            'e': (4, lambda h: float(binascii.hexlify(h), 16)),
-            'g': (4, lambda h: float(binascii.hexlify(h), 16)),
+        size_formatter_info = {
+            's' : -1,
+            'c' : 1,
+            'i' : 4,
+            'd' : 4,
+            'u' : 4,
+            'x' : 4,
+            'hh': 1,
+            'h' : 2,
+            'l' : 4,
+            'll': 8,
+            'f' : 8,
+            'g' : 8,
+            'e' : 8,
         }
-        formatters = re.findall("%(?:\d+\$)?([cdifosuxXhlL]+)", self._message)
+        type_formatter_info = {
+            'c' : 'U{}',
+            'i' : 'MSB_I{}',
+            'd' : 'MSB_I{}',
+            'u' : 'MSB_U{}',
+            'f' : 'MSB_D{}',
+            'e' : 'MSB_D{}',
+            'g' : 'MSB_D{}',
+            'x' : 'MSB_U{}',
+        }
+
+        formatters = re.findall("%(?:\d+\$)?([cdieEfgGosuxXhlL]+)", self._message)
 
         cur_byte_index = 0
         data_chunks = []
 
         for f in formatters:
-            format_size, format_func = formatter_info[f]
+            # If the format string we found is > 1 character we know that a length
+            # field is included and we need to adjust our sizing accordingly.
+            f_size_char = f_type = f[-1]
+            if len(f) > 1:
+                f_size_char = f[:-1]
+
+            fsize = size_formatter_info[f_size_char.lower()]
 
             try:
-                # Normal data chunking is the current byte index + the size
-                # of the relevant data type for the formatter
-                if format_size > 0:
-                    end_index = cur_byte_index + format_size
+                if f != 's':
```

Comment:
It probably makes more sense to check if `f_type == 's'` here since we're using `f` to seed that value. It shouldn't change anything if left as is, but probably clearer updated.